ctru-rs/src/services/ndsp/wave.rs

```diff
@@ -0,0 +1,138 @@
+use ctru_sys::ndspChnWaveBufClear;
+
+use super::AudioFormat;
+use crate::linear::LinearAllocator;
+
+/// Base struct to represent audio wave data. This requires audio format information.
+pub struct WaveBuffer {
+    /// Buffer data. This data must be allocated on the LINEAR memory.
+    data: Box<[u8], LinearAllocator>,
+    audio_format: AudioFormat,
+    nsamples: usize, // We don't use the slice's length here because depending on the format it may vary
+                     // adpcm_data: AdpcmData, TODO: Requires research on how this format is handled.
+}
+
+/// Informational struct holding the raw audio data and playaback info. This corresponds to [ctru_sys::ndspWaveBuf]
+pub struct WaveInfo<'b> {
+    /// Data block of the audio wave (plus its format information).
+    buffer: &'b mut WaveBuffer,
+    // Holding the data with the raw format is necessary since `libctru` will access it.
+    pub(crate) raw_data: ctru_sys::ndspWaveBuf,
+    played_on_channel: Option<u8>,
```

Comment:
Should this be an `Option<super::Channel>`? Or is that a different type of channel than the one used here?

Reply:
Indeed. Depending on how I'll solve the `Channel` lifetime issue it could be hard to hold it in there, so I see no reason to hold anything more than a single byte.